### PR TITLE
Refactor to use requests json

### LINKED ISSUE #2852: Replace all urllib / urllib2 imports and uses with `requests`
`urllib2` is old, and a bit clunky -- I've found recently it can't do some of things I needed it to do, and it also causes problem for moving to Python3, and generally seems to make the code more complex.

Instead of re-using it, and adding new features that use the older mechanisms, can we consider refactoring to use `requests` instead?

My PR where I stripped out `urllib2` and `StringIO` etc to enable `HEAD` requests:  #2838

a similar PR that went with using `requests` to avoid old bugs: #2779 

By using `requests` (rather than adding imports from `six`) we should be getting a better tool for making requests, and reducing code complexity, and the number of imports. 

Relates to #2308 since a common use of `simplejson` is to parse `urllib2` responses. `requests` has a builtin `json()` method so an extra import is not required.

### PR DESCRIPTION
**Pull request by aasifkhan7**
Closes #2852 
related to #3694 

@cclauss clubbed json requests method in this PR.

### CODE REVIEW

**Comment by cclauss** [comment]
Please fix the following https://travis-ci.org/github/internetarchive/openlibrary/jobs/727821530#L600-L605

**Comment by aasifkhan7** [comment]
@cclauss All Travis checks passed.

File: openlibrary/catalog/amazon/add_covers.py [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
+import requests
 
 base = 'http://ia331526.us.archive.org:7001/openlibrary.org/log/'
 
 out = open('edition_and_isbn', 'w')
 offset = '2009-06-01:0'
 while not offset.startswith('2010-03-17:'):
     url = base + offset
-    ret = simplejson.load(urlopen(url))
+    ret = requests.get(url).json()
```
Comment (mekarpeles) [inline openlibrary/catalog/amazon/add_covers.py:10]: tested add cover

Comment (mekarpeles) [reply openlibrary/catalog/amazon/add_covers.py:10]: https://dev.openlibrary.org/works/OL20600W/Gulliver's_Travels

File: openlibrary/plugins/upstream/borrow.py [hunk 2]
```diff
@@ -566,7 +567,7 @@ def get_loan_status(resource_id):
 
     url = '%s/is_loaned_out/%s' % (loanstatus_url, resource_id)
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/borrow.py:568]: Tested borrows w/ https://dev.openlibrary.org/books/OL25128089M/What_technology_wants

File: openlibrary/plugins/upstream/models.py [hunk 3]
```diff
@@ -719,8 +719,7 @@ def get_covers(self, offset=0, limit=20):
 
         try:
             url = '%s/b/query?cmd=ids&olid=%s' % (get_coverstore_url(), ",".join(olids))
-            data = urllib.request.urlopen(url).read()
-            cover_ids = simplejson.loads(data)
+            cover_ids = requests.get(url).json()
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/models.py:722]: We confirmed covers loading for subjects

**Review by mekarpeles — APPROVED** [review]
lgtm, tested on dev w/ @cclauss

**Pull request merged**